apps/server/src/controllers/auth.controller.ts

```
@@ -38,38 +36,34 @@ export const registerUser = async (
 
     const saltRounds = 10;
     const password_hash = await bcrypt.hash(password, saltRounds); // hash the password saltRound number of times
-    const { data, error } = await supabase
-      .from('users')
-      .insert([{ username, password_hash }]) // data variable name must correspond to the table column name
-      .select('id, username, created_at, default_currency_code');
-
-    if (error) {
-      // check for unique constraint violation (Supabase error code for unique violation is '23505')
-      if (error.code === '23505') {
-        res
-          .status(409)
-          .json({ message: 'Username already taken.', error: error.message });
-        return;
-      }
-      console.log('Error creating user', error);
-      res.status(500).json({
-        message: 'Internal server error. Please try again later.',
-      });
-    }
-    if (!data || data.length === 0) {
-      console.error(
-        'User creation returned no data, though no explicit error.',
-      );
-      res.status(500).json({ message: 'User created, but no data returned.' });
-      return;
-    }
+
+    const newUser = await prisma?.users.create({
```

Why do we need the `?` here? If prisma was indeed Null, I think we should throw an error loudly

---

Yeah. It shouldn't be there. Removed